continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanCondition.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Krzysztof Slusarski
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.krzysztofslusarski.asyncprofiler.mbean;
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+class ContinuousAsyncProfilerMBeanCondition implements Condition {
+    @Override
+    public boolean matches(ConditionContext conditionContext, AnnotatedTypeMetadata annotatedTypeMetadata) {
+        String repositoryType = conditionContext.getEnvironment().getProperty("async-profiler.continuous.properties-repository");
+        return "jmx".equalsIgnoreCase(repositoryType);
```

Comment:
I wonder if we should somehow respect Spring Boot default value of `spring.jmx.enabled=false`. What is your opinion?

Reply:
I thought about that but:
- I believe that it is only Spring Boot property
- I remember that I created JMX server on older Spring without that flag
- it is (unfortunately) common pattern, that libraries have their own "jmx" properties, eg: hikari